tests/metadata/generate.rs

```diff
@@ -0,0 +1,262 @@
+        .unwrap();
+
+    let hash = targets
+        .as_ref()
+        .targets()
+        .get(&VirtualTargetPath::new(step.to_string().into()).unwrap())
+        .unwrap()
+        .hashes()
+        .get(&HashAlgorithm::Sha256)
+        .unwrap();
+
+    let target_str = if consistent_snapshot {
+        format!("{}.{}", hash, step.to_string())
+    } else {
+        step.to_string()
+    };
+    let target_path = TargetPath::new(target_str.into()).unwrap();
+    repo.store_target(target_data, &target_path).await.unwrap();
+
+    let version = if consistent_snapshot {
+        MetadataVersion::Number((step + 1).into())
+    } else {
+        MetadataVersion::None
+    };
+
+    repo.store_metadata(&targets_path, &version, &targets)
+        .await
+        .unwrap();
+
+    let snapshot_path = MetadataPath::from_role(&Role::Snapshot);
+    let snapshot = SnapshotMetadataBuilder::new()
+        .insert_metadata(&targets, &[HashAlgorithm::Sha256])
+        .unwrap()
+        .signed::<Json>(&keys.get("snapshot").unwrap())
+        .unwrap();
+
+    repo.store_metadata(&snapshot_path, &version, &snapshot)
+        .await
+        .unwrap();
+
+    let timestamp_path = MetadataPath::from_role(&Role::Timestamp);
+    let timestamp = TimestampMetadataBuilder::from_snapshot(&snapshot, &[HashAlgorithm::Sha256])
+        .unwrap()
+        .signed::<Json>(&keys.get("timestamp").unwrap())
+        .unwrap();
+
+    // Timestamp doesn't require a version even in consistent_snapshot.
+    repo.store_metadata(&timestamp_path, &MetadataVersion::None, &timestamp)
+        .await
+        .unwrap();
+}
+
+async fn generate_repos(dir: &str, consistent_snapshot: bool) -> tuf::Result<()> {
+    // Create initial repo.
+    // TODO: @erickt, this impl is very tied to the json keys file. Do we need it to be more flexible?
+    let json_keys = init_json_keys(KEYS_PATH);
+    let mut keys = init_role_keys(&json_keys);
+    let dir0 = Path::new(dir).join("0");
+    let repo = FileSystemRepositoryBuilder::new(dir0)
+        .metadata_prefix(Path::new("repository"))
+        .targets_prefix(Path::new("repository").join("targets"))
+        .build()?;
+
+    update_root(&repo, &keys, None, 1, consistent_snapshot).await;
+    add_target(&repo, &keys, 0, consistent_snapshot).await;
+
+    let mut i: u8 = 1;
+    let rotations = vec![
+        Some(Role::Root),
+        Some(Role::Targets),
+        Some(Role::Snapshot),
+        Some(Role::Timestamp),
+        None,
+    ];
+    for r in rotations.iter() {
+        // Initialize new repo and copy the files from the previous step.
+        let dir_i = Path::new(dir).join(i.to_string());
+        let repo = FileSystemRepositoryBuilder::new(dir_i)
+            .metadata_prefix(Path::new("repository"))
+            .targets_prefix(Path::new("repository").join("targets"))
+            .build()
+            .unwrap();
+        copy_repo(dir, i);
+
+        let root_signer: Option<PrivateKey> = match r {
```

Comment:
Nit: I think you can leave off `Option<PrivateKey>`.

Reply:
Done.